### ISSUE TITLE
White screen with error in TextualEvent component

### ISSUE THREAD

**Issue by jryans** [vector-im/element-web#11029]
From @anoadragon453:

```
Unhandled rejection TypeError: cli is null
    getMembersWithUser@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:142222:20
    roomMembers@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:142245:31
    _callee$@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:142112:74
From previous event:
    _showMembersAccordingToMembershipWithLL@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:142133:25
    componentWillMount@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:142022:18
    callComponentWillMount@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:395806:14
    mountClassInstance@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:395899:27
    updateClassComponent@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:399424:23
    beginWork$1@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:400943:16
    beginWork$$1@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:405631:14
    performUnitOfWork@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:404649:12
    workLoopSync@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:404623:22
    renderRoot@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:404316:11
    runRootCallback@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:403992:20
    flushSyncCallbackQueueImpl/<@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:393791:24
    unstable_runWithPriority@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:413232:12
    runWithPriority$2@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:393743:10
    flushSyncCallbackQueueImpl@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:393787:24
    flushSyncCallbackQueue@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:393776:3
    scheduleUpdateOnFiber@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:403869:9
    enqueueSetState@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:395538:17
    ./node_modules/react/cjs/react.development.js/Component.prototype.setState@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:408639:16
    _viewRoom/<@https://chat.amorgan.xyz/bundles/a0a28d5d17576cf48ea2/bundle.js:82696:20
```

**Comment by jryans** [comment]
@anoadragon453, do you have a good way to reproduce this, or is intermittent?

**Comment by jryans** [comment]
`TypeError: cli is null` is not the original problem, this is just a symptom: when some React component throws an error at the moment, this unmounts the app and stops the client, which leads to `cli` being null.

At the time the client is stopped in the logs, we see:

```
2019-10-02T09:50:11.235Z E The above error occurred in the <TextualEvent> component:
    in TextualEvent (created by EventTile)
    in div (created by EventTile)
    in div (created by EventTile)
    in EventTile (created by MessagePanel)
    in li (created by MessagePanel)
    in ol (created by ScrollPanel)
    in div (created by ScrollPanel)
    in div (created by AutoHideScrollbar)
    in div (created by AutoHideScrollbar)
    in AutoHideScrollbar (created by ScrollPanel)
    in ScrollPanel (created by MessagePanel)
    in MessagePanel (created by TimelinePanel)
    in TimelinePanel (created by RoomView)
    in div (created by RoomView)
    in div (created by RoomView)
    in div (created by MainSplit)
    in MainSplit (created by RoomView)
    in main (created by RoomView)
    in RoomView (created by LoggedInView)
    in div (created by LoggedInView)
    in DragDropContext (created by LoggedInView)
    in div (created by LoggedInView)
    in LoggedInView (created by MatrixChat)
    in MatrixChat
```

however the logs do not contain any detail of what the "above" error actually was... 😕

**Comment by anoadragon453** [comment]
A bit random. My entire Riot was slowing down, so I tried to clear cache and the same thing occurred.

I then realized my media store was timing out every request, which caused Riot to chug, perhaps because it had made and was waiting on hundreds of media requests for all of my room icons and user avatars?

So perhaps it's just a performance issue and React dies because of it.

**Comment by jryans** [comment]
If you see it again, look for any errors near the `stopping MatrixClient` line... I am wondering if there's some error in the console that isn't being captured to the debug logs somehow.

**Comment by kittykat** [comment]
@anoadragon453 have you seen this issue recently?

**Comment by anoadragon453** [comment]
I can't recall seeing `TypeError: cli is null` at any point recently, so I assume this bug is no longer relevant.

Thanks for checking in!
